src/nt_users_tool/excel_processing.py

```diff
@@ -0,0 +1,82 @@
+from openpyxl import Workbook, load_workbook
+from datetime import date
+
+from constants import SHEET_ALL_USERS, SHEET_EXPIRED, SHEET_EXPIRES_SOON, SHEETS_NAME_LIST, COLUMNS_LIST, MONTHS_TO_EXPIRE, TWO_YEAR_GAP
+from net_commands import NTUserInfo
+
+def read_nt_users(worksheet) ->list:
+    """Generates a list of nt_user found in the worksheet.
+
+    :param worksheet: The worksheet with nt_users inside.
+    :return: A list of nt_user (string).
+    """
+    nt_user_id_list = []
+    for row in worksheet.rows:
+        for cell in row:
+            if cell.value != None:
+                nt_user_id_list.append(cell.value)
+    return nt_user_id_list
+
+def create_results_sheets(workbook: Workbook):
+    """Creates the sheets inside the given workbook according to parameter
+
+    :param workbook: The workbook in which to add the sheets.
+    """
+    for sheet_name in SHEETS_NAME_LIST:
+        workbook.create_sheet(sheet_name)
+
+def fill_one_row(worksheet, row_number: int, columns: list, nt_user_info: NTUserInfo):
+    """Fills row_number of columns on the given worksheet, with the info in nt_user_info.
+
+    :param worksheet: The worksheet that needs to be changed.
+    :type worksheet: an openpyxl worksheet object.
+    :param row_number: the row at which the changes happen.
+    :param columns: the range of columns we write on.
+    :param user_info: The information that we are writing.
+    """
+    for i, column in enumerate(columns):
+        worksheet[column + str(row_number)] = nt_user_info[i]
+
+
+def fill_all_sheets(workbook: Workbook, nt_user_info_list: list):
```

Comment:
here you can be more precise about the type hint
```Python
nt_user_info_list: list[NTUserInfo]
```
It helps the IDE understand what you are passing to the function which can then help you debug way more easilly :)

Reply:
![image](https://user-images.githubusercontent.com/98102969/197147759-4b47a1aa-37be-4d92-b920-e9a2e46f280a.png)

Reply:
Fixed in 638ef90.

Reply:
yes, you can only do that in Python 3.10
With older versions you have to do
```Python
from typing import List
List[MyElement]
```